From 3eb3e9c98da6d431e3b38d82f9601db7deb8b2a5 Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sun, 2 May 2021 14:56:58 +0200
Subject: [PATCH] [mobile] rename default_pin -> user_pass

Prepare to add a config option to allow non-numeric passwords, by giving
the existing default_pin screen the more generic name user_pass.
Adjust the title of the screen too.

"default" in the file name was referring to configuring the default
user's PIN/password as opposed to the SSH user. However, I think
replacing it with "user" makes it more intuitive.
---
 modules/mobile/mobile.qml                        | 10 +++++-----
 modules/mobile/mobile.qrc                        |  2 +-
 .../mobile/{default_pin.qml => user_pass.qml}    | 16 ++++++++--------
 3 files changed, 14 insertions(+), 14 deletions(-)
 rename modules/mobile/{default_pin.qml => user_pass.qml} (85%)

diff --git a/modules/mobile/mobile.qml b/modules/mobile/mobile.qml
index 3821345..6977061 100644
--- a/modules/mobile/mobile.qml
+++ b/modules/mobile/mobile.qml
@@ -19,7 +19,7 @@ Page
         "welcome": null, /* titlebar disabled */
         "install_target": "Installation target",
         "install_target_confirm": "Warning",
-        "default_pin": "Lockscreen PIN",
+        "user_pass": "User password",
         "ssh_confirm": "SSH server",
         "ssh_credentials": "SSH credentials",
         "fs_selection": "Root filesystem",
@@ -33,8 +33,8 @@ Page
          "screens": ["welcome"]},
         {"name": "installTarget",
          "screens": ["install_target", "install_target_confirm"]},
-        {"name": "userPin",
-         "screens": ["default_pin"]},
+        {"name": "userPassword",
+         "screens": ["user_pass"]},
         {"name": "sshd",
          "screens": ["ssh_confirm", "ssh_credentials"]},
         {"name": "fsType",
@@ -46,7 +46,7 @@ Page
     ]
     property var featureIdByScreen: (function() {
         /* Put "features" above into an index of screen name -> feature id:
-         * featureIdByScreen = {"welcome": 0, "default_pin": 1, ...} */
+         * featureIdByScreen = {"welcome": 0, "user_pass": 1, ...} */
         var ret = {};
         for (var i=0; i<features.length; i++) {
             for (var j=0; j<features[i]["screens"].length; j++) {
@@ -253,7 +253,7 @@ Page
         return true;
     }
 
-    /* Input validation: user-screens (default_pin, ssh_credentials) */
+    /* Input validation: user-screens (user_pass, ssh_credentials) */
     function validatePin(userPin, userPinRepeat, errorText) {
         var pin = userPin.text;
         var repeat = userPinRepeat.text;
diff --git a/modules/mobile/mobile.qrc b/modules/mobile/mobile.qrc
index a35eed6..27e592a 100644
--- a/modules/mobile/mobile.qrc
+++ b/modules/mobile/mobile.qrc
@@ -7,7 +7,7 @@
   <file>install_target.qml</file> <!-- install from external to internal? -->
   <file>install_target_confirm.qml</file> <!-- overwrite internal storage? -->
 
-  <file>default_pin.qml</file> <!-- default user: pin -->
+  <file>user_pass.qml</file> <!-- default user: password -->
   <file>ssh_confirm.qml</file> <!-- sshd: enable or not? -->
   <file>ssh_credentials.qml</file> <!-- sshd user: username, password -->
   <file>fs_selection.qml</file> <!-- filesystem selection -->
diff --git a/modules/mobile/default_pin.qml b/modules/mobile/user_pass.qml
similarity index 85%
rename from modules/mobile/default_pin.qml
rename to modules/mobile/user_pass.qml
index f8e8487..a6bfba0 100644
--- a/modules/mobile/default_pin.qml
+++ b/modules/mobile/user_pass.qml
@@ -33,11 +33,11 @@ Item {
     }
 
     TextField {
-        id: userPin
+        id: userPass
         anchors.top: description.bottom
         placeholderText: qsTr("PIN")
         echoMode: TextInput.Password
-        onTextChanged: validatePin(userPin, userPinRepeat, errorText)
+        onTextChanged: validatePin(userPass, userPassRepeat, errorText)
         text: config.userPassword
 
         /* Let the virtual keyboard change to digits only */
@@ -54,12 +54,12 @@ Item {
     }
 
     TextField {
-        id: userPinRepeat
-        anchors.top: userPin.bottom
+        id: userPassRepeat
+        anchors.top: userPass.bottom
         placeholderText: qsTr("PIN (repeat)")
         inputMethodHints: Qt.ImhDigitsOnly
         echoMode: TextInput.Password
-        onTextChanged: validatePin(userPin, userPinRepeat, errorText)
+        onTextChanged: validatePin(userPass, userPassRepeat, errorText)
         text: config.userPassword
 
         anchors.horizontalCenter: parent.horizontalCenter
@@ -68,7 +68,7 @@ Item {
     }
 
     Text {
-        anchors.top: userPinRepeat.bottom
+        anchors.top: userPassRepeat.bottom
         id: errorText
         visible: false
         wrapMode: Text.WordWrap
@@ -86,8 +86,8 @@ Item {
 
         text: qsTr("Continue")
         onClicked: {
-            if (validatePin(userPin, userPinRepeat, errorText)) {
-                config.userPassword = userPin.text;
+            if (validatePin(userPass, userPassRepeat, errorText)) {
+                config.userPassword = userPass.text;
                 navNext();
             }
         }
-- 
2.26.3

